python/ray/experimental/channel/common.py

```diff
@@ -77,6 +77,7 @@ def create_channel(
         self,
         writer: Optional["ray.actor.ActorHandle"],
         reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]],
+        num_shm_buffers: Optional[int] = None,
```

Comment:
At channel constructor, we can just all it `num_buffers` to be more general and not referring to a specific implementation of shared memory. We can still assert this is None for nccl.

Reply:
Instead of this, I moved it to a constructor arg of `SharedMemoryType`, so it is only applied to shm channel. I currently don't really see the need of buffer in nccl. We can change it in the future if needed